### PR TITLE
GVRAssetLoader APIs for custom loading

### PR DESCRIPTION
**Pull request by NolaDonato**
Added APIs to GVRAssetLoader which take a GVRResourceVolume instead of a String to describe the asset to load. This allows callers to subclass GVRResourceVolume to provide custom loading behavior. Modified GVRResourceVolume to keep the filename it was intially created with.

GearVRF DCO signed off by: Nola Donato nola.donato@samsung.com

### CODE REVIEW

**Comment by liaxim** [comment]
gvr-3dcursor-simple exits with this exception:

```
02-13 08:57:28.451 22292 22409 W System.err: java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.String.length()' on a null object reference
02-13 08:57:28.461 22292 22409 W System.err: 	at org.gearvrf.GVRResourceVolume.<init>(GVRResourceVolume.java:102)
02-13 08:57:28.461 22292 22409 W System.err: 	at org.gearvrf.GVRAssetLoader.loadModel(GVRAssetLoader.java:1193)
02-13 08:57:28.461 22292 22409 W System.err: 	at org.gearvrf.GVRContext.loadModel(GVRContext.java:660)
02-13 08:57:28.461 22292 22409 W System.err: 	at org.gearvrf.io.cursorsimple.CursorMain.onInit(CursorMain.java:87)
02-13 08:57:28.461 22292 22409 W System.err: 	at java.lang.reflect.Method.invoke(Native Method)
02-13 08:57:28.461 22292 22409 W System.err: 	at org.gearvrf.GVREventManager.invokeMethod(GVREventManager.java:307)
02-13 08:57:28.461 22292 22409 W System.err: 	at org.gearvrf.GVREventManager.sendEventWithMaskParamArray(GVREventManager.java:126)
02-13 08:57:28.461 22292 22409 W System.err: 	at org.gearvrf.GVREventManager.sendEventWithMask(GVREventManager.java:113)
02-13 08:57:28.461 22292 22409 W System.err: 	at org.gearvrf.GVREventManager.sendEvent(GVREventManager.java:108)
02-13 08:57:28.461 22292 22409 W System.err: 	at org.gearvrf.GVRViewManager$1$1.run(GVRViewManager.java:320)
02-13 08:57:28.461 22292 22409 W System.err: 	at android.os.Handler.handleCallback(Handler.java:739)
02-13 08:57:28.461 22292 22409 W System.err: 	at android.os.Handler.dispatchMessage(Handler.java:95)
02-13 08:57:28.461 22292 22409 W System.err: 	at android.os.Looper.loop(Looper.java:158)
02-13 08:57:28.461 22292 22409 W System.err: 	at android.os.HandlerThread.run(HandlerThread.java:61)
```

**Comment by liaxim** [comment]
Similar exception in gvr-jassimp.

**Comment by liaxim** [comment]
gvr-lua doesn't show anything. Nothing obvious in the logs.

**Comment by NolaDonato** [comment]
When I run gvr-lua it shows a black screen for a while. Then it shows astroboy walking with "GVRf scripting in Lua". The test does not require internet or storage permissions - it loads astroboy from the assets directory. I ran my test on the S7 Edge. Which phone was used when gvr-lua did not show anything:?

**Comment by liaxim** [comment]
Note 5. I will be rerunning the demos with your updates and will pay special attention to gvr-lua.

File: GVRf/Framework/framework/src/main/java/org/gearvrf/GVRAssetLoader.java [hunk 1]
```diff
@@ -1014,81 +1019,110 @@ public GVRModelSceneObject loadScene(String filePath, GVRScene scene) throws IOE
...
     {
-        AssetRequest assetRequest = new AssetRequest(model, filePath, scene, true);
-        String ext = filePath.substring(filePath.length() - 3).toLowerCase();
+        Threads.spawn(new Runnable()
+        {
+            public void run()
+            {
+                AssetRequest assetRequest = new AssetRequest(model, volume, scene, true);
+                String filePath = volume.getFileName();
+                String ext = filePath.substring(filePath.length() - 3).toLowerCase();
```
Comment (liaxim) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/GVRAssetLoader.java]: I think the assumption the file extension is always three characters long may be incorrect. It is something that can wait though.

Comment (liaxim) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/GVRAssetLoader.java]: Looks like it is meant just for x3d so nevermind.

**Comment by liaxim** [comment]
Looks good. Will merge Friday morning if there are no other comments.

**Pull request merged**